### DISCUSSION TITLE
twitch chat intergration with sources

### DISCUSSION THREAD

**Discussion by InkedTay** [WarmUpTill/SceneSwitcher#1243] [Q&A]
ive managed to wrap my head around the rest of the automatic scene switcher pretty well however with the twitch choice is triggered im having trouble making it read chat well. The idea is to make a source pop on screen when the chatbot makes a certain predetermined command. it crashed multiple times while trying it with different variations of the command however i made the chat register that i had typed in a simple 'lol' and it worked once and tried to set back to the original command and got another crash. im sorry if im missing something simple. im quite new to all the github and obs etc
[Crash 2024-10-15 22-51-15.txt](https://github.com/user-attachments/files/17388955/Crash.2024-10-15.22-51-15.txt)
[Crash 2024-10-15 22-34-39.txt](https://github.com/user-attachments/files/17388956/Crash.2024-10-15.22-34-39.txt)
[Crash 2024-10-15 22-31-44.txt](https://github.com/user-attachments/files/17388958/Crash.2024-10-15.22-31-44.txt)

**Comment by InkedTay** [comment]
Forgive me im unsure what magic you managed. I couldnt get it working for the life of me but as soon as i sent the crash report it stopped crashing on me. although i am curious as to what happened if you have any idea


UPDATE after getting one command to work and attempting to set another up we have gotten an additional crash 
https://obsproject.com/logs/KzDtWQFFFv6slzHV

**Comment by WarmUpTill** [comment]
>im sorry if im missing something simple. im quite new to all the github and obs etc

Sorry about the crashes!
That is definitively not your fault!
And thank you for reporting the issue!

Which release of the plugin do you have installed?
In which scenario are you experiencing the crashes? (When configuring the settings for the Advanced Scene Switcher or just when it is running without changing any settings?)

The most recent beta release has a few bug fixes / crash fixes related to the Twitch condition, which might potentially resolve your issue.
You can find it here:
https://github.com/WarmUpTill/SceneSwitcher/releases/tag/1.28.0-beta1

Hopefully this will already resolve your issues,
If not please do let me know and I will try to figure out what is going wrong.

**Reply by InkedTay** [reply]
I have now updated to the beta release of your plugin and i was running the default version of obs. i was able to replicate the crash with previous settings in the beta version but i think it was a user error not understanding what i was having it do. The exact scenario is after setting the macro in scene switcher (im in offstream twitch chat to run tests if that matters) i go over to the stream chat dock and type it in against the proper REGEX ie. dammitbeans whereas the proper was DAMMITBEANS [i had everything toggled because i couldnt get it to pick up chat at all initially] (theoretically a random chatter just types the word i dont want it to trigger but for exact spelling and caps from my chatbot it would) and when i would type in the lower case nothing would happen so id wait a few seconds and do the full caps version and obs would crash and if it was a fresh macro itll wipe it.... in the middle of typing this and trying to explain i realized triggered a ton of different stuff that mightve been confusing it and making it crash. i now turned off the clear message buffer and turned off the different regex toggles and it seems to work now. Is there anything i can do to help understand what i was doing wrong to help me wrap my head around?

Thanks for your speedy response and fantastic product by the way i have been battling so many different small tasks and projects trying to go from head to paper learning obs and stuff (youre program has also helped me create a perpetual dice machine for my brb screen)

**Reply by InkedTay** [reply]
Am i possibly doing something incorrect regarding the use of spaces in the chat description box? any of my commands that include spaces dont get picked up. like i can type "as;dlfkjsalkfjladsf dammitbeans asldkfjsalkdjf;lsakdjff" in and itll work just fine but if the command is dammit beans itll never read it in chat

**Reply by InkedTay** [reply]
and furthering the specific instance of the crashes seem to be around the "clear message buffer when matching message was found" when that toggles it seems to not want to read the chat well and then i test it again to see if i did something wrong and the second instance makes the crash occur.
Ive got my use case fixed just wanted to try to be helpful

**Reply by WarmUpTill** [reply]
>Ive got my use case fixed just wanted to try to be helpful

That is good to hear and thank you very much!
I think I was able to track down the problem.

A build with a fix will be available here in a few minutes:
https://github.com/WarmUpTill/SceneSwitcher/actions/runs/11389064762?pr=1245
You will have to scroll down to the artifacts section while being logged into GitHub to be able to download it.
![](https://github.com/user-attachments/assets/03fd589d-75a3-4bf5-ae41-a898d6626be1)
